src/diagram/Timeline.ts

```diff
@@ -0,0 +1,92 @@
+import { Cancellable } from "../command/Commands";
+import { Diagram, Memento } from "./Diagram";
+
+export class Snapshot<T extends Cancellable> {
+  constructor(readonly origin: Memento, readonly modifier: T) {}
+}
+
+export class Timeline<T extends Cancellable> {
+  private diagram!: Diagram | null;
```

Comment:
why use an exclamation mark here?

Reply:
Removed, the variable diagram has been replaced by the MainApp's diagram for this child class.